### PR TITLE
CI: Add a Github workflow to run php linting

### PR DESCRIPTION
**Pull request by anomiex**
#### Changes proposed in this Pull Request:

We seem to be moving away from Travis and to Github Actions. So, this
moves linting and PHPCompatibility out of Travis and into a Github
Action.

This also adds running of the full PHPCS Standard over the non-excluded
files, so we don't regress on them in the future as we have in the past.

#### Jetpack product discussion

None outside of some private Slack chats.

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Is the new workflow running? Does it pass?

* Look at #17720, which is a dummy PR based on this one that adds some files with errors. Verify that the results there are as expected.

#### Proposed changelog entry for your changes:

* Developer only, none needed.

### CODE REVIEW

**Comment by kraftbj** [comment]
What's the plan with the failing linting tests? Both look to be due to Composer installation issues; do we need to do something special to get them to pass?

File: .github/workflows/php-linting.yml
Comment (kraftbj) [inline .github/workflows/php-linting.yml:94]: Should we run the remove before the install?

**Comment by anomiex** [comment]
> What's the plan with the failing linting tests? Both look to be due to Composer installation issues; do we need to do something special to get them to pass?

Needs a master merge to pick up 3ae959f5a is all.

File: .github/workflows/php-linting.yml
Comment (anomiex) [reply .github/workflows/php-linting.yml:94]: The remove does an implicit update, which can wind up installing wrong versions of packages (cf. #17259). IIRC a `composer install` also avoids the problem by more closely sticking to the lock file.

We could do the remove first with the `COMPOSER_ROOT_VERSION=dev-master` that was done in #17259 if you really want.

**Review by kraftbj — APPROVED** [review]

File: .github/workflows/php-linting.yml
Comment (jeherve) [inline .github/workflows/php-linting.yml:51]: Should we keep "PHP Lint" in the name here? Since the UI already shows the action name before, we may not need to display that twice?

image

Comment (anomiex) [reply .github/workflows/php-linting.yml:51]: It includes the workflow name in some places, but not others.

ss1
ss2

Comment (jeherve) [reply .github/workflows/php-linting.yml:51]: Ah, good to know.

**Comment by anomiex** [comment]
Needs re-approval after parent patch merge.

**Review by kraftbj — APPROVED** [review]

**Pull request merged**
